File: gsdishopping_app/lib/models/campaign.dart
/// Campanha do shopping (ex.: "Semana da Moda"), vinda do GSDIShoppingApi
/// (`GET /api/catalog/campaigns`). Agrupa os [Coupon]s que o cliente pode
/// ver enquanto ela estiver ativa.
class Campaign {
  const Campaign({
    required this.id,
    required this.title,
    required this.description,
    this.imageUrl,
    this.startsAt,
    this.endsAt,
  });

  final int id;
  final String title;
  final String description;
  final String? imageUrl;
  final DateTime? startsAt;
  final DateTime? endsAt;

  factory Campaign.fromJson(Map<String, dynamic> json) => Campaign(
        id: json['id'] as int,
        title: json['title'] as String,
        description: json['description'] as String,
        imageUrl: json['imageUrl'] as String?,
        startsAt: json['startsAt'] != null ? DateTime.tryParse(json['startsAt'] as String) : null,
        endsAt: json['endsAt'] != null ? DateTime.tryParse(json['endsAt'] as String) : null,
      );
}

File: GSDIShoppingApi/Models/Campaign.cs
namespace GSDIShoppingApi.Models;

/// <summary>
/// Campanha do shopping (ex.: "Semana da Moda") — agrupa um ou mais
/// <see cref="Coupon"/> que o cliente pode ver/resgatar enquanto a
/// campanha estiver ativa. Diferente de <see cref="Promotion"/>: campanha
/// é sempre do shopping como um todo (não de uma loja específica) e existe
/// para organizar cupons, não para anunciar um desconto pontual.
/// </summary>
public class Campaign
{
    public int Id { get; set; }
    public required string Title { get; set; }
    public required string Description { get; set; }
    public string? ImageUrl { get; set; }
    public DateTime? StartsAt { get; set; }
    public DateTime? EndsAt { get; set; }
    public DateTime CreatedAt { get; set; } = DateTime.UtcNow;

    public ICollection<Coupon> Coupons { get; set; } = new List<Coupon>();
}
